src/rdkafka.h

```diff
@@ -9848,6 +9869,221 @@ RD_EXPORT void rd_kafka_DeleteAcls(rd_kafka_t *rk,
+
+/**
+ * @brief Get the rd_kafka_topic_partition_result_t object at index \p idx from
+ * the rd_kafka_topic_partition_result_t array.
+ *
+ * @param result The array of rd_kafka_topic_partition_result_t objects.
+ * @param idx The index of the object to be retrieved.
+ *
+ * @returns the rd_kafka_topic_partition_result_t object at index \p idx from
+ * the rd_kafka_topic_partition_result_t array.
+ *
+ * @remark The index \p idx must be less than the size of the array.
+ *
+ *
+ */
+RD_EXPORT const rd_kafka_topic_partition_result_t *
+rd_kafka_topic_partition_result_by_idx(
+    const rd_kafka_topic_partition_result_t **result,
+    size_t idx);
+
+/**
+ * @brief Destroy and free a rd_kafka_topic_partition_result_t object.
+ *
+ * @param partition_result The rd_kafka_topic_partition_result_t object to be
+ * destroyed.
+ */
+RD_EXPORT void rd_kafka_topic_partition_result_destroy(
+    rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @brief Destroy and free an array of rd_kafka_topic_partition_result_t
+ * objects.
+ *
+ * @param partition_results The array of rd_kafka_topic_partition_result_t
+ * objects to be destroyed.
+ * @param partition_results_cnt The number of elements in the array.
+ *
+ * @remark The array itself is not freed.
+ */
+RD_EXPORT void rd_kafka_topic_partition_result_destroy_array(
+    rd_kafka_topic_partition_result_t **partition_results,
+    int32_t partition_results_cnt);
+
+/**
+ * @brief Get error code from the ElectLeaders result.
+ *
+ * @param result The ElectLeaders result event.
+ *
+ * @returns the error code from the ElectLeaders result.
+ */
+RD_EXPORT rd_kafka_resp_err_t
+rd_kafka_ElectLeadersResult_error(const rd_kafka_ElectLeadersResult_t *result);
+
+/**
+ * @brief Get the ElectLeaders result from the ElectLeaders result event.
+ *
+ * @param result The ElectLeaders result event.
+ *
+ * @returns the ElectLeaders result from the ElectLeaders result event.
+ *
+ * @remark This error is related to the overall ElectLeaders operation.
+           Individual partition related errors are present at partition level.
+ */
+
+RD_EXPORT const rd_kafka_ElectLeadersResult_t *
+rd_kafka_ElectLeaders_result(const rd_kafka_ElectLeaders_result_t *result);
+
+/**
+ * @brief Destroy and free an rd_kafka_ElectLeadersResult_t object.
+ *
+ * @param result The rd_kafka_ElectLeadersResult_t object to be destroyed.
+ */
+
+RD_EXPORT void
+rd_kafka_ElectLeadersResult_destroy(rd_kafka_ElectLeadersResult_t *result);
+
+/**
+ * @brief Get the array of rd_kafka_topic_partition_result_t objects from the
+ *        ElectLeaders result event and populates the size of the
+ *        array in \p cntp.
+ *
+ * @param result The ElectLeaders result event.
+ * @param cntp The number of elements in the array.
+ *
+ * @returns the array of rd_kafka_topic_partition_result_t objects from the
+ *         ElectLeaders result event.
+ */
+RD_EXPORT const rd_kafka_topic_partition_result_t **
+rd_kafka_ElectLeadersResult_partitions(
```

Comment:
Rearrange APIs for the same struct in such a way that constructor is present first then getter and setters and then destructors. Check older APIs as well for understanding.